Abandon panics on nil Done channel

If no Done channel is set, Abandon can panic:

```
2015/08/14 05:09:27 client 2 close()

2015/08/14 05:09:27 Client [2] sent abandon signal to request[messageID = 1]

panic: close of nil channel

goroutine 1607 [running]:

github.com/vjeantet/ldapserver.(*Message).Abandon(0xc20a5a3bc0)

	/home/travis/gopath/src/github.com/openshift/origin/Godeps/_workspace/src/github.com/vjeantet/ldapserver/message.go:54 +0x2c

created by github.com/vjeantet/ldapserver.(*client).close

	/home/travis/gopath/src/github.com/openshift/origin/Godeps/_workspace/src/github.com/vjeantet/ldapserver/client.go:166 +0x3e0
```

--- a/message.go
+++ b/message.go
@@ -51,7 +51,9 @@ func (m *Message) String() string {
 // Abandon close the Done channel, to notify handler's user function to stop any
 // running process
 func (m *Message) Abandon() {
-	close(m.Done)
+	if m.Done != nil {
+		close(m.Done)
+	}
 }
 
 //GetDoneChannel return a channel, which indicate the the request should be

